src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php

```diff
@@ -1591,7 +1592,7 @@ private function addDefaultParametersMethod(): string
             $export = $this->exportParameters([$value], '', 12, $hasEnum);
             $export = explode('0 => ', substr(rtrim($export, " ]\n"), 2, -1), 2);
 
-            if ($hasEnum || preg_match("/\\\$container->(?:getEnv\('(?:[-.\w\\\\]*+:)*+\w++'\)|targetDir\.'')/", $export[1])) {
+            if ($hasEnum || preg_match("/\\\$container->(?:getEnv\('(?:[-.\w\\\\]*+:)*+\w*+'\)|targetDir\.'')/", $export[1])) {
```

Comment:
this change allows to derivate dynamic parameters from other ones, eg `%env(key:foo:default:bar:)%` extracts key `foo` from parameter `bar` (which is supposed to be an array here)